src/interactions/panZoomInteraction.ts

```diff
@@ -93,12 +93,14 @@ export module Interactions {
       var newCornerDistance = this._cornerDistance();
 
       if (this._xScale != null && newCornerDistance !== 0 && oldCornerDistance !== 0) {
-        PanZoom._magnifyScale(this._xScale, oldCornerDistance / newCornerDistance, oldCenterPoint.x);
-        PanZoom._translateScale(this._xScale, oldCenterPoint.x - newCenterPoint.x);
+        var invertedPX = this._xScale.invert(oldCenterPoint.x);
+        this._xScale.zoomLevel(this._xScale.zoomLevel() * oldCornerDistance / newCornerDistance);
+        PanZoom._translateScale(this._xScale, this._xScale.scale(invertedPX) - newCenterPoint.x);
```

Comment:
`_magnifyScale(...)` was a private static method to pull out the calculations for calculating the new domain of the scale.  Now that we actually have an endpoint on the scale that will do the calculations inherently, I didn't see the need of `_magnifyScale(...)`.